spec/rubocop/cop/rspec/multiple_memoized_helpers_spec.rb

```diff
@@ -0,0 +1,162 @@
+# frozen_string_literal: true
+
+RSpec.describe RuboCop::Cop::RSpec::MultipleMemoizedHelpers, :config do
+  let(:cop_config) { { 'Max' => 1 } }
+
+  it 'flags excessive `#let`' do
+    expect_offense(<<~RUBY)
+      describe Foo do
+      ^^^^^^^^^^^^^^^ Example group has too many memoized helpers [2/1]
+        let(:foo) { Foo.new }
+        let(:bar) { Bar.new }
+      end
+    RUBY
+  end
+
+  it 'flags excessive `#let!`' do
+    expect_offense(<<~RUBY)
+      describe Foo do
+      ^^^^^^^^^^^^^^^ Example group has too many memoized helpers [2/1]
+        let(:foo) { Foo.new }
+        let!(:bar) { Bar.new }
+      end
+    RUBY
+  end
+
+  it 'ignores a reasonable number of memoized helpers' do
+    expect_no_offenses(<<~RUBY)
+      describe Foo do
+        let(:foo) { Foo.new }
+      end
+    RUBY
+  end
+
+  it 'ignores overridden `#let`' do
+    expect_no_offenses(<<~RUBY)
+      describe Foo do
+        let(:foo) { Foo.new }
+        context do
+          let(:foo) { Foo.new(1) }
+          context do
+            let(:foo) { Foo.new(1, 2) }
+          end
+        end
+      end
+    RUBY
+  end
+
+  it 'ignores `#subject`' do
+    expect_no_offenses(<<~RUBY)
+      describe Foo do
+        subject(:bar) { Bar.new }
+        let(:foo) { Foo.new }
+      end
+    RUBY
+  end
+
+  it 'ignores `#subject!`' do
+    expect_no_offenses(<<~RUBY)
+      describe Foo do
+        subject!(:bar) { Bar.new }
+        let(:foo) { Foo.new }
+      end
+    RUBY
+  end
+
+  it 'ignores `#subject` without a name' do
+    expect_no_offenses(<<~RUBY)
+      describe Foo do
+        subject { Foo.new }
+        let(:bar) { Bar.new }
+      end
+    RUBY
+  end
+
+  it 'flags nested `#let`' do
+    expect_offense(<<~RUBY)
+      describe Foo do
+        let(:foo) { Foo.new }
+
+        context 'when blah' do
+        ^^^^^^^^^^^^^^^^^^^^^^ Example group has too many memoized helpers [2/1]
+          let(:bar) { Bar.new }
+        end
+      end
+    RUBY
+  end
+
+  it 'ignores distributed `#let`' do
+    expect_no_offenses(<<~RUBY)
+      describe Foo do
+        context 'when bloo' do
+          let(:foo) { Foo.new }
+        end
+
+        context 'when blah' do
+          let(:bar) { Bar.new }
+        end
+      end
+    RUBY
+  end
+
+  context 'when using AllowSubject configuration', :config do
```

Comment:
Perhaps it would be clearer to write “when disabling AllowSubject configuration”?